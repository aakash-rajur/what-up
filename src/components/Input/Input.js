import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Input extends Component {
    static propTypes = {
        onValueReturn: PropTypes.func
    };

    constructor(props) {
        super(props);
        this.onReturn = this.onReturn.bind(this);
    }

    render() {
        let inputProps = {...this.props};
        delete inputProps.onValueReturn;
        return (
            <input {...inputProps} onKeyUp={this.onReturn}/>
        )
    }

    onReturn(e) {
        let {
            keyCode,
            target: {value}
        } = e, {
            onKeyUp,
            onValueReturn
        } = this.props;
        onKeyUp && onKeyUp(e);
        keyCode === 13 && onValueReturn && onValueReturn(value)
    }
}

export default Input;
