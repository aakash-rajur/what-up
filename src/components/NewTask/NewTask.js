import PropTypes from 'prop-types'
import React from 'react';
import {withNewTaskAddition} from "../../utils/apollo";
import Input from "../Input/Input";

export function NewTask({newTask, onChange, addNewTask, ...rest}) {
	let inputProps = {...rest};
	delete inputProps.onDone;
	return (
		<Input value={newTask} onChange={onChange}
		       onValueReturn={addNewTask} {...inputProps}/>
	);
}

NewTask.propTypes = {
	newTask: PropTypes.string,
	onChange: PropTypes.func
};

export default withNewTaskAddition(NewTask);
