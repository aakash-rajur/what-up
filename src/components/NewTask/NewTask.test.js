import React from 'react';
import Input from "../Input/Input";
import {NewTask} from './NewTask';

describe('NewTask component', () => {
	let newTask = 'helloWorld',
		onChange = stub(),
		addNewTask = stub(),
		props = {
			newTask,
			onChange,
			addNewTask,
			type: 'text',
			autoFocus: true,
			className: 'lorem',
			placeholder: 'ipsum'
		},
		component = shallow(<NewTask {...props}/>);
	
	it('dom integrity', checkDOM(component, [{
		selector: Input,
		props: {...props}
	}]));
	
	matchSnapshot(component);
});
