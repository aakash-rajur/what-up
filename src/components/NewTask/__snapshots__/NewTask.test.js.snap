// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTask component match snapshot 1`] = `
<Input
  autoFocus={true}
  className="lorem"
  onChange={[Function]}
  onValueReturn={[Function]}
  placeholder="ipsum"
  type="text"
  value="helloWorld"
/>
`;
